packages/jest-util/src/deepCyclicCopy.ts

```diff
@@ -23,9 +28,20 @@ export default function deepCyclicCopy<T>(
     return cycles.get(value);
   } else if (Array.isArray(value)) {
     return deepCyclicCopyArray(value, options, cycles);
-  } else {
-    return deepCyclicCopyObject(value, options, cycles);
+  } else if (isMap(value)) {
+    return deepCyclicCopyMap(value, options, cycles);
+  } else if (isSet(value)) {
+    return deepCyclicCopySet(value, options, cycles);
+  } else if (Buffer.isBuffer(value)) {
```

Comment:
Please forgive the long delay on reviewing your updates. If I understand correctly, the reason for a deep copy of collections is because a member value might need to be replaced by an asymmetric matcher. Therefore, is it necessary to deep copy the types Buffer, Date, NumberArray, and RegExp type? Because they cannot contain an asymmetric matcher as an item or property value, true?

Reply:
Yes, you are right!
But if not doing copy correctly, it make Date, RegExp  missing some property and method or causing NumberArray with wrong structure.
For example, copying NumberArray:
```js
UInt8Array [1, 2] // expected
UInt8Array { '0': 1, '1': 2 } // copy by original deepCyclicCopy
```

If I skip to copy those type, just return their value. The implementation conflict with function named `deepCyclicCopy`.

Or create a new function for only copying necessary object in replacing asymmetric matcher process.
What do you think?

Reply:
Yes to your suggestion create a new function:

1. Undo changes to `deepCyclicCopy.ts` in `jest-utils` package, which I think solves any problems with browser build and resolve the questions that Simen asked a few days ago

2. Decide on a descriptive name and create a new file in `jest-matcher-utils` package:

    * the `blacklist` option is not needed, correct?
    * always keep prototype, correct?
    * copy collections whose values might be replaced, and as you said, return other objects

Your coding style is very clear to read and so my mind is now catching up to your innovating solution.